### PR TITLE
8330670: Obsolete ScavengeBeforeFullGC

### PR DESCRIPTION
**Pull request by albertnetymk**
Obsolete a gc product flag.

Test: tier1-3

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8330671](https://bugs.openjdk.org/browse/JDK-8330671) to be approved

### Issues
 * [JDK-8330670](https://bugs.openjdk.org/browse/JDK-8330670): Obsolete ScavengeBeforeFullGC (**Enhancement** - P4)
 * [JDK-8330671](https://bugs.openjdk.org/browse/JDK-8330671): Obsolete ScavengeBeforeFullGC (**CSR**)

### Reviewers
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**)
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18861/head:pull/18861` \
`$ git checkout pull/18861`

Update a local copy of the PR: \
`$ git checkout pull/18861` \
`$ git pull https://git.openjdk.org/jdk.git pull/18861/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18861`

View PR using the GUI difftool: \
`$ git pr show -t 18861`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18861.diff">https://git.openjdk.org/jdk/pull/18861.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18861#issuecomment-2066426399)

### CODE REVIEW

File: src/hotspot/share/gc/z/zDriver.cpp [hunk 1]
```diff
@@ -310,8 +310,7 @@ static bool should_preclean_young(GCCause::Cause cause) {
   // catch that above and perform young gen pre-cleaning.
   assert(!should_clear_soft_references(cause), "Clearing soft references without pre-cleaning young gen");
 
-  // Preclean young if implied by configuration
-  return ScavengeBeforeFullGC;
+  return false;
```
Comment (stefank) [inline src/hotspot/share/gc/z/zDriver.cpp:313]: I'm not sure if this is correct. We previously returned true here if the user didn't set ScavengeBeforeFullGC, right? Maybe we need a ZGC-specific flag to allow this "precleaning" to be turned off?

**Review by stefank — CHANGES_REQUESTED** [review]

File: src/hotspot/share/gc/z/zDriver.cpp (hunk 1 above)
Comment (stefank) [reply src/hotspot/share/gc/z/zDriver.cpp:313]: Ahh, I missed that. Nowadays we typically put that kind of code in the GC-specific places and not in the shared code.

**Review by stefank — APPROVED** [review]

**Review by tschatzl — APPROVED** [review]

**Comment by albertnetymk** [comment]
Thanks for review.

/integrate

**Pull request closed without merging**